Marlin/src/feature/bedlevel/bdl/bdl.cpp

```diff
@@ -127,16 +128,14 @@ void BDS_Leveling::process() {
             babystep.set_mm(Z_AXIS, cur_z - z_sensor);
             DEBUG_ECHOLNPGM("BD:", z_sensor, ", Z:", cur_z, "|", current_position.z);
           }
-          else {
-            babystep.set_mm(Z_AXIS, 0);   //if (old_cur_z <= cur_z) Z_DIR_WRITE(HIGH);
-            //stepper.apply_directions();   // TODO: Remove this line as probably not needed
-          }
+          else
+            babystep.set_mm(Z_AXIS, 0);
         }
       #endif
 
       old_cur_z = cur_z;
       old_buf_z = current_position.z;
-      endstops.bdp_state_update(z_sensor <= 0.01f);
+      endstops.bdp_state_update(z_sensor <= abs(probe.offset.z));
```

Comment:
This seems to imply that the accuracy of the probe goes down in proportion to the nozzle-to-probe-triggerpoint distance. But, the probe-trigger distance is not exactly *directly proportional* to the nozzle-to-probe-triggerpoint distance since the probe position (above the nozzle) will vary depending on how it is mounted. Maybe this should be configured as a "`BD_SENSOR_TOLERANCE`" configuration option instead, so users can choose how strict the BD Sensor is about calling `bdp_state_update`.

Reply:
OK, I will configure a `BD_SENSOR_HOME_POSITION` for users to choose home position.

Reply:
"Home position" for Z is understood as being where the nozzle touches the bed – which is to say, Z0 – so the name "`BD_SENSOR_HOME_Z_POSITION`" may be confusing. We could get questions like "Does a sensor have a home position?" and "How do I home the sensor?"

Reply:
The name `BD_SENSOR_HOME_Z_POSITION` maybe not properly, what I mean is that the nozzle does not touch the bed while homing Z, but if it touches the bed it is zero. 
For example, 
here the `BD_SENSOR_HOME_Z_POSITION` is 0.8, the nozzle will stop moving after the BDsensor measure the distance is <=0.8mm ,  since the  `bdp_state_update ` is update in the main loop that is not in real time and the actual stop position maybe 0.75mm, so I add this `current_position.z = bdl.read();` in the function ` set_axis_is_at_home() ` that will let the `current_position.z` is 0.75mm

Reply:
So let me see if I understand how this relates to `probe.offset.z`. Please correct my understanding if I have anything wrong here….
- A typical probe has a trigger height, usually something like 1-2mm below the nozzle, and so in that case we might have a `probe.offset.z` (`M851 Z`) of -1.5 and that is added to the current Z position when the probe triggers so that it gives us a usable mesh value relative to the nozzle position.
- However, the BD Sensor produces a continuous reading to indicate how far it is from the bed. So, if the BD Sensor is mounted at 2mm above the nozzle, it would give a reading close to 2mm when the nozzle is touching the bed (assuming the bed is flat).
- Therefore, we subtract 2mm from the reading to get the height value relative to the nozzle, and that value can subsequently be used as the nozzle-to-bed distance.
- Any probe distance can be used for `G28` homing, because all we need is some reliable nozzle-to-bed distance to know where the bed is in relation to the nozzle.
- So the constant `BD_SENSOR_HOME_Z_POSITION` represents the maximum BD Sensor distance reading that we will treat as the "trigger point" of the sensor, so that anything below that probe-to-bed distance is considered to be triggered.

What height reading does the BD Sensor report when it is at 1mm above the surface? Does it report 1mm or is there some distance that it always centers around, so that it actually gives a negative value when at that height?

I guess what I want to understand is how we use `probe.offset.z` (`M851 Z`) with the BD Sensor, and can we use `probe.offset.z` instead of `BD_SENSOR_HOME_Z_POSITION`, so that it is adjustable with `M851 Z`, or is it already adjustable with `M851 Z`? Is `BD_SENSOR_HOME_Z_POSITION` already taking into account the `probe.offset.z` and just providing a height below which the Z endstop/probe is considered "triggered" for the benefit of things (like `G28`) that need a definitive endstop (`M119`) state rather than just a distance reading? If we don't currently use `probe.offset.z` because it doesn't fit with the BD Sensor idiom, perhaps it should be repurposed and used in place of `BD_SENSOR_HOME_Z_POSITION`.

Reply:
Thanks!
one thing I did not described clearly is that: after we mount the BD sensor, we need to do calibration at first, so no matter the BDsensor is mounted at 1mm or 2mm, the nozzle touching the bed is 0mm after calibration.

Reply:
`BD_SENSOR_HOME_Z_POSITION`   just providing a height below which the Z endstop is considered "triggered" only benefit the G28 and M119, and no effort to other probe like G29.